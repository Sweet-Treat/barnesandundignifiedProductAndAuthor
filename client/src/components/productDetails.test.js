import 'jsdom-global/register'; //at the top of file , even  , before importing react
import ProductDetails from './productDetails';
import React from 'react';
import { mount} from 'enzyme';



describe ('ProductDetails React test', () => {
  it ('render ProductDetails', (done) => {
    const record = { 'isbn13': '9781524763169',
      'author': 'Barack Obama',
      'bookTitle': 'A Promised Land',
      'publisherName': 'Crown Publishing Group',
      'publisherLink': '/publisher',
      'publicationDate': '11/17/2020',
      'series': '',
      'bookCategory': 'Autobiography',
      'ageRange': '',
      'pages': 768,
      'salesRank': 1,
      'format': '',
      'editionDescription': '',
      'productDimensions': '6.40(w) x 9.40(h) x 1.50(d)'
    };
    const wrapper = mount(<ProductDetails record={record}/>);
    //console.log("wrapper", wrapper.debug());
    expect(wrapper.debug()).toMatchSnapshot();
    expect(wrapper.containsMatchingElement(<tr>
      <th>
        ISBN-13:
      </th>
      <td>
        9781524763169
      </td>
    </tr>)).toBeTruthy();
    expect(wrapper.containsMatchingElement(<tr>
      <th>
        Publisher:
      </th>
      <td>
        <a href="/publisher">
          Crown Publishing Group
        </a>
      </td>
    </tr>)).toBeTruthy();
    done();
  });
});

