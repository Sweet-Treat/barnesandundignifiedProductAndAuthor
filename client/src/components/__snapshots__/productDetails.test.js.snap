// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails React test render ProductDetails 1`] = `
"<ProductDetails record={{...}}>
  <div className=\\"wrapper\\">
    <div className=\\"slide\\">
      <h2>
        Product Details
      </h2>
      <table>
        <tbody>
          <tr>
            <th>
              ISBN-13: 
            </th>
            <td>
              9781524763169
            </td>
          </tr>
          <tr>
            <th>
              Publisher: 
            </th>
            <td>
              <a href=\\"/publisher\\">
                Crown Publishing Group
              </a>
            </td>
          </tr>
          <tr>
            <th>
              Publication Date: 
            </th>
            <td>
              11/17/2020
            </td>
          </tr>
          <tr>
            <th>
              Pages: 
            </th>
            <td>
              768
            </td>
          </tr>
          <tr>
            <th>
              Sales Rank: 
            </th>
            <td>
              1
            </td>
          </tr>
          <tr>
            <th>
              Product dimensions: 
            </th>
            <td>
              6.40(w) x 9.40(h) x 1.50(d)
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</ProductDetails>"
`;
